### DISCUSSION TITLE
How to bind the right mouse button

### DISCUSSION THREAD

**Discussion by lautnerxing** [TomSchimansky/CustomTkinter#1717] [Q&A] [answered]
I have a tag that I want to have some other selectable tags when I right-click on it, but I can't find how to bind the right mouse button

**Comment by lautnerxing** [comment]
For example, I want to achieve this effect:
![image](https://github.com/TomSchimansky/CustomTkinter/assets/51226709/b80a35d3-0d44-4ac6-ab3f-da3fba90dff9)

**Comment by abhaymakes** [comment]
You can try this:

```python
your_widget.bind("<Button-2>", function_you_want_to_call)
```

**Reply by avalon60** [reply]
I usually use a lambda, as in most cases I find I need to pass parameters. 
E.g.

`geometry_widget.bind("<Button-3>", lambda event, l_widget=geometry_widget: deselect_widget(widget_id=l_widget))`

**Reply by abhaymakes** [reply]
Yes, in case you need parameters in your function to be passed, refer to @avalon60's comment.

**Reply by avalon60** [reply]
At the same time, keep it as simple as possible. If you don't need to pass parameters, go with @TheFallen-Cat's suggestion.

**Reply by lautnerxing** [reply]
Thank you, but I have one more question. It seems that customkinter does not have a function to draw a tree diagram.
As for the binding issue, I have already resolved it:
![image](https://github.com/TomSchimansky/CustomTkinter/assets/51226709/171a0623-0e7d-49c9-9aee-0e6281267a07)

**Reply by avalon60** [reply]
Not in the base CustomTkinter package. I haven't used it personally,  but have you checked this out:
https://github.com/Akascape/TkNodeSystem

**Reply by lautnerxing** [reply]
I have found out that customkinter does not support treeview yet   ~O.o~...

**Reply by ghost** [reply]
@Akascape has created many reliable, and working extension widgets for CustomTkinter. I'd suggest trying out the node system @avalon60 linked.

**Reply by Akascape** [reply, marked as answer]
@lautnerxing 
I guess you wanted a treeview and not a node tree diagram.

These may help:

https://github.com/TomSchimansky/CustomTkinter/discussions/524#discussioncomment-3898307

https://github.com/TomSchimansky/CustomTkinter/discussions/1483

**Comment by urlan** [comment]
I'm sorry, but I think CustomTkinter doesn't have this feature. You need to use Tkinter instead with _Menu_, for example:

https://www.youtube.com/watch?v=KRuUtNxOb_k

**Reply by JanPanthera** [reply]
Yeah it would be awesome to have a real tree view directly within customtkinter, but its more fun to create your own, right? 😹😹
![image](https://github.com/TomSchimansky/CustomTkinter/assets/20288289/735f9493-638f-4370-833f-9a1ff7d78bf4)
